lock-manager/src/lock-operations.ts

```diff
@@ -73,21 +77,20 @@ class DatabaseStorage implements Storage {
     }
   }
 
-  async release(releaseReq: ReleaseLocks, db: Db): Promise<Array<String>> {
+  async release(releaseReq: ReleaseLocks, db: Db): Promise<Array<string>> {
     logger.debug("release keys: %s", JSON.stringify(releaseReq))
-    let { lockIds: keys, owner } = releaseReq
     const query = {
       name: "delete-key",
       text: `DELETE FROM locks WHERE lock_id = ANY ($1) AND
         EXISTS (SELECT lock_id FROM locks WHERE lock_id = ANY ($1) ) AND lock_metadata ->> 'lockOwner' = $2 RETURNING lock_id`,
-      values: [ keys, owner ],
+      values: [ releaseReq.lockIds, releaseReq.owner ],
     }
 
     const result = await db.execute(query)
 
     let unlockedKeys = result?.rows?.map(({ lock_id }) => lock_id)
     if (unlockedKeys?.length === 0) {
-      throw new Error("release(): No valid lock and owner combination found in database to delete")
+      throw new Error("No valid lock and owner combination found in database to delete")
```

Comment:
Question: Should this be treated as an error?

Reply:
It does not matter as long as caller knows what to do. For example if user1 not owning a lock but trying to release it, this will produce error because "user1 + lockid" is not known. Similar with keep alive. service. Using errors gives us just 2 states: error or success. If we do not want to use error then we ned three states, which is harder to code maybe :)

Reply:
I'd say that releasing a previously held lock (by deleting an existing record) and having no lock to release yield the same outcomes for the caller. They are effectively idempotent. This shouldn't be treated as an error.